ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -221,6 +314,41 @@ private ExecutorPlans planQuery(
     return new ExecutorPlans(logicalPlan, physicalPlan);
   }
 
+  private LogicalPlanNode buildAndValidateLogicalPlan(
+      final ConfiguredStatement<?> statement,
+      final ImmutableAnalysis analysis,
+      final KsqlConfig config
+  ) {
+    final OutputNode outputNode = new LogicalPlanner(config, analysis, engineContext.getMetaStore())
+        .buildPlan();
+    return new LogicalPlanNode(
+        statement.getStatementText(),
+        Optional.of(outputNode)
+    );
+  }
+
+  private PullPhysicalPlan buildPullPhysicalPlan(
+      final KsqlExecutionContext ksqlEngine,
+      final LogicalPlanNode logicalPlan,
+      final KsqlConfig config,
+      final ImmutableAnalysis analysis,
```

Comment:
This analysis exposes all of the parser expressions.  Isn't that something that should be at the logical planner layer?  I would assume that the physical planner would take some logical nodes which have all of the relevant data extracted from the expressions.

Maybe this is just a first step, and there's still some intermixing of layers, but I just thought I would ask.

Reply:
The analyzer performs some initial validation and categorization and is a different step happening before the logical plan. The logical planner exposes as well parser expressions but inside logical nodes. That's what I want to change. So that at the physical planner we don't deal with any AST expressions anymore, only logical entities.